From 6471d69ca4758bb1be8c8eccf7d0a2f98a184b5c Mon Sep 17 00:00:00 2001
From: Christoph Cullmann <christoph@cullmann.dev>
Date: Sun, 18 Jan 2026 21:19:47 +0100
Subject: [PATCH] fix binary compatibility

don't remove virtual functions

one needs do use

xxx_BUILD_DEPRECATED_SINCE

and not

xxx_ENABLE_DEPRECATED_SINCE
---
 src/filewidgets/kdiroperator.h | 4 ++--
 src/widgets/jobuidelegate.h    | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/filewidgets/kdiroperator.h b/src/filewidgets/kdiroperator.h
index 692ba878da..43fa0571b9 100644
--- a/src/filewidgets/kdiroperator.h
+++ b/src/filewidgets/kdiroperator.h
@@ -573,7 +573,7 @@ public:
      */
     bool followSelectedDirectories() const;
 
-#if KIOFILEWIDGETS_ENABLE_DEPRECATED_SINCE(6, 15)
+#if KIOFILEWIDGETS_BUILD_DEPRECATED_SINCE(6, 15)
     /*!
      * Starts and returns a KIO::DeleteJob to delete the given \a items.
      *
@@ -650,7 +650,7 @@ public:
      */
     virtual void setDropOptions(int options);
 
-#if KIOFILEWIDGETS_ENABLE_DEPRECATED_SINCE(6, 15)
+#if KIOFILEWIDGETS_BUILD_DEPRECATED_SINCE(6, 15)
     /*!
      * Starts and returns a KIO::CopyJob to trash the given \a items.
      *
diff --git a/src/widgets/jobuidelegate.h b/src/widgets/jobuidelegate.h
index b9d89be477..9298ad140a 100644
--- a/src/widgets/jobuidelegate.h
+++ b/src/widgets/jobuidelegate.h
@@ -86,7 +86,7 @@ public:
      */
     static void unregisterWindow(QWidget *window);
 
-#if KIOWIDGETS_ENABLE_DEPRECATED_SINCE(6, 15)
+#if KIOWIDGETS_BUILD_DEPRECATED_SINCE(6, 15)
     /*!
      * Ask for confirmation before deleting/trashing \a urls.
      *
-- 
GitLab

